File: Algorithms/src/696. Count Binary Substrings/Code/solution.go
func countBinarySubstrings(s string) int {
    
    prevGroup := 0
    currGroup := 1
    result := 0
    
    for i := 1; i < len(s); i++ {
        
        if s[i] == s[i-1] {
            currGroup++
        } else {
            if prevGroup < currGroup {
                result += prevGroup
            } else {
                result += currGroup
            }
            
            prevGroup = currGroup
            currGroup = 1
        }
    }
    
    if prevGroup < currGroup {
        result += prevGroup
    } else {
        result += currGroup
    }
    
    return result
}

File: Algorithms/src/696. Count Binary Substrings/Code/solution.py
class Solution:
    def countBinarySubstrings(self, s: str) -> int:
        
        prevGroup = 0
        currGroup = 1
        result = 0
        
        for i in range(1, len(s)):
            
            if s[i] == s[i - 1]:
                currGroup += 1
            else:
                result += min(prevGroup, currGroup)
                prevGroup = currGroup
                currGroup = 1
        
        result += min(prevGroup, currGroup)
        
        return result
